lib/neo4j/http/batch_node_client.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Neo4j
+  module Http
+    class BatchNodeClient < NodeClient
+      def upsert_node(node)
+        raise "#{node.key_name} value cannot be blank - (node keys: #{node.to_h.keys})" if node.key_value.blank?
+
+        cypher = <<-CYPHER
+          MERGE (node:#{node.label} {#{node.key_name}: $key_value})
+          ON CREATE SET node += $attributes
+          ON MATCH SET node += $attributes
+          return node
+        CYPHER
+
+        {
+          statement: cypher,
+          parameters: { key_value: node.key_value, attributes: node.attributes }
+        }
+      end
```

Comment:
My only issue with this approach is that the CYPHEr that we use for "batch" vs non-batch should be the same, but this creates that in 2 places now.  I'd prefer if we this wasn't a "client" as much as a "CypherBuilder" that was reused in the regular client, and used to compose cypher in the batch case

Reply:
@rstawarz I have extracted the cypher building to its own method in the clients and then called them from the single and batch client. The cypher is defined once and we now inherit instead of duplicate